lms/static/scripts/frontend_apps/components/BookSelector.js

```diff
@@ -0,0 +1,156 @@
+import {
+  IconButton,
+  SvgIcon,
+  TextInput,
+  TextInputWithButton,
+  Thumbnail,
+} from '@hypothesis/frontend-shared';
+import { createElement } from 'preact';
+import { useRef, useState } from 'preact/hooks';
+
+import { useService, VitalSourceService } from '../services';
+
+/**
+ * @typedef {import('../api-types').Book} Book
+ */
+
+/**
+ * @typedef BookSelectorProps
+ * @prop {Book|null} selectedBook
+ * @prop {(b: Book|null) => void} onSelectBook - Callback invoked when user selects a book
+ */
+
+/**
+ * Component that prompts a user to enter/paste a URL to a VitalSource book.
```

Comment:
```suggestion
 * Component that prompts a user to paste a URL to a VitalSource book.
```

We basically don't expect the user to ever enter a URL here manually. I suppose they might select it from an autocomplete list of recent values after focusing the field, if the browser offers.